lib/components/canvas/canvas.dart

```diff
@@ -37,10 +43,53 @@ class Canvas extends StatelessWidget {
 
   final void Function(EditorImage image)? setAsBackground;
 
-  final bool currentToolIsSelect;
+  final Tool currentTool;
   final double currentScale;
   final bool placeholder;
 
+  int toolToOnyx(Tool currentTool) {
+    if (placeholder) return 5;
+    if (currentTool is Pencil) {
+      return 3;
+    } else if (currentTool is Highlighter) {
+      return 4;
+    } else if (currentTool is Eraser) {
+      return 1;
+    } else if (currentTool is Select) {
+      return 1;
+    } else if (currentTool is LaserPointer) {
+      return 1;
+    } else if (currentTool is Pen) {
+      if (currentTool.isPressureEnabled()) {
+        return 2;
+      } else {
+        return 1;
+      }
+    } else {
+      return 5;
+    }
+  }
+
+  int getColor() {
+      if (currentTool is Pen) {
+        return (currentTool as Pen).color.toARGB32();
+      } else {
+        return Colors.black.toARGB32();
+      }
+  }
+  double getWidth() {
+      if (currentTool is Pen) {
+        double baseSize = (currentTool as Pen).getSize() * currentScale;
+        if ((currentTool as Pen).isPressureEnabled()) {
+          return baseSize;
+        } else {
+          return baseSize * 2;
+        }
+      } else {
+        return 3;
+      }
+  }
```

Comment:
I think these methods would be best named `getOnyxTool`, `getOnyxColor`, `getOnyxWidth` so their purpose is more clear